packages/core-data/src/hooks/use-entity-records.ts

```diff
@@ -77,9 +84,13 @@ export default function __experimentalUseEntityRecords< RecordType >(
 	const queryAsString = addQueryArgs( '', queryArgs );
 
 	const { data: records, ...rest } = useQuerySelect(
-		( query ) =>
-			query( coreStore ).getEntityRecords( kind, name, queryArgs ),
-		[ kind, name, queryAsString ]
+		( query ) => {
+			if ( ! options.__experimentalEnabled ) {
+				return {};
```

Comment:
Is it expected to return an empty object rather than `null` here? It seems weird to me 🤔 , I'd imagine that it at least returns an empty array than an object.

Reply:
If this was `null`, then the destructuring would error. Returning an empty object makes the `data` undefined in line 86.  Perhaps this could return an object like `{ data: [] }`, but I don't think it matters that much – it's just placeholder value for when the hook is disabled.

Reply:
Actually, reconsidering it – if you had to ask that question then it reads confusing. I turned it into
```js
	if ( ! options.enabled ) {
		return {
			data: [],
		};
	}
```

If only to make it more readable.